projects/fsastorefrontlib/src/cms-components/agent/agent-search-list/agent-search-list.component.ts

```diff
@@ -26,16 +29,21 @@ export class AgentSearchListComponent implements OnInit, OnDestroy {
     this.subscription.add(
       this.route.queryParams.subscribe(params => this.initialize(params))
     );
-
     this.searchResults$ = this.agentSearchService.getResults().pipe(
       tap(agents => {
         if (agents) {
+          this.initialActiveAgent = agents.agents[0];
           this.pagination = {
             currentPage: agents.pagination.page,
             pageSize: agents.pagination.count,
             totalPages: agents.pagination.totalPages,
             totalResults: agents.pagination.totalCount,
           };
+          // Reset searchQuery in case user search agent from the List view clicking on the location icon
+          // This ensures getting appropriate list by clicking on Back to list button
+          if (this.initialActiveAgent?.email === this.searchQuery) {
```

Comment:
I don't get this one. This will reset searchQuery only if customers search by email. What if the customer searches by name and we again have this initial agent, should we reset it again when we click back?